openpype/hosts/maya/plugins/inventory/connect_geometry.py

```diff
@@ -0,0 +1,139 @@
+from maya import cmds
+
+from openpype.pipeline import InventoryAction, get_representation_context
+
+
+class ConnectGeometry(InventoryAction):
+    """Connect geometries within containers.
+
+    Source container will connect to the target containers, by searching for
+    matching geometry IDs (cbid).
+    Source containers are of family; "animation" and "pointcache".
+    The connection with be done with a live world space blendshape.
+    """
+
+    label = "Connect Geometry"
+    icon = "link"
+    color = "white"
+
+    def process(self, containers):
+        # Validate selection is more than 1.
+        message = (
+            "Only 1 container selected. 2+ containers needed for this action."
+        )
+        if len(containers) == 1:
+            self.display_warning(message)
+            return
+
+        # Categorize containers by family.
+        containers_by_family = {}
+        for container in containers:
+            family = get_representation_context(
+                container["representation"]
+            )["subset"]["data"]["family"]
+            try:
+                containers_by_family[family].append(container)
+            except KeyError:
+                containers_by_family[family] = [container]
+
+        # Validate to only 1 source container.
+        source_containers = containers_by_family.get("animation", [])
+        source_containers += containers_by_family.get("pointcache", [])
+        source_container_namespaces = [
+            x["namespace"] for x in source_containers
+        ]
+        message = (
+            "{} animation containers selected:\n\n{}\n\nOnly select 1 of type "
+            "\"animation\" or \"pointcache\".".format(
+                len(source_containers), source_container_namespaces
+            )
+        )
+        if len(source_containers) != 1:
+            self.display_warning(message)
+            return
+
+        source_object = source_containers[0]["objectName"]
+
+        # Collect matching geometry transforms based cbId attribute.
+        target_containers = []
+        for family, containers in containers_by_family.items():
+            if family in ["animation", "pointcache"]:
+                continue
+
+            target_containers.extend(containers)
+
+        source_data = self.get_container_data(source_object)
+        matches = []
+        node_types = []
```

Comment:
`node_types` should really be a `set`. We're only using it to do a lookup later to see whether `xgmPalette` is one of the nodes types.

The same goes for `nodes_types` in the `get_container_data` result.

Reply:
Sure, can do